src/trans.rs

```diff
@@ -719,6 +727,48 @@ impl<'v, 'tcx: 'v> BinaryenFnCtxt<'v, 'tcx> {
                 }
             }
 
+            Rvalue::CheckedBinaryOp(ref op, ref left, ref right) => {
+                // TODO: This shouldn't just be a copy BinaryOp above!
+                // We should do some actual _checking_!
```

Comment:
On second thought, this would be much easier to do with a call to an emulation function, and letting binaryen (or downstream jit compilers) inline that as appropriate.  Is there precedent for doing that sort of thing?